File: backend/src/main/java/mx/codemx/cursos/model/Modulo.java
package mx.codemx.cursos.model;

import jakarta.persistence.*;

@Entity
@Table(name = "modulos", schema = "cursos")
public class Modulo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private String icono;

    @Column(name = "orden", nullable = false)
    private Integer orden;

    public Modulo() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }

    public String getIcono() { return icono; }
    public void setIcono(String icono) { this.icono = icono; }

    public Integer getOrden() { return orden; }
    public void setOrden(Integer orden) { this.orden = orden; }
}

File: backend/Modules/Cursos/Modulo.cs
namespace CodeMX.Api.Modules.Cursos;

/// <summary>Módulo de aprendizaje (curso).</summary>
public class Modulo
{
    public long Id { get; set; }
    public string Titulo { get; set; } = "";
    public string Descripcion { get; set; } = "";
    public string Icono { get; set; } = "";
    public int Orden { get; set; }
}
